safenode/src/bin/kadnode.rs

```diff
@@ -1,233 +1,178 @@
-                    //         //     println!(
-                    //         //         "Successfully put record {:?}",
-                    //         //         std::str::from_utf8(key.as_ref()).unwrap()
-                    //         //     );
-                    //         // }
-                    //         // QueryResult::PutRecord(Err(err)) => {
-                    //         //     eprintln!("Failed to put record: {err:?}");
-                    //         // }
-                    //         // QueryResult::StartProviding(Ok(AddProviderOk { key })) => {
-                    //         //     println!(
-                    //         //         "Successfully put provider record {:?}",
-                    //         //         std::str::from_utf8(key.as_ref()).unwrap()
-                    //         //     );
-                    //         // }
-                    //         // QueryResult::StartProviding(Err(err)) => {
-                    //         //     eprintln!("Failed to put provider record: {err:?}");
-                    //         // }
-                    //         _ => {
-                    //             //
-                    //         }
-                    //     }
-                    // }
-                    _ => {}
                 }
-
             }
         }
     });
 
-    sender
-}
-
-#[tokio::main]
-async fn main() -> Result<()> {
-    let log_dir = grab_log_dir();
-    let _log_appender_guard = init_node_logging(&log_dir)?;
-
-    info!("start");
-    let channel = run_swarm();
-
-    let x = xor_name::XorName::from_content(b"some random content here for you");
-
-    if let Err(e) = channel.send(SwarmCmd::Search(x)).await {
-        debug!("Error while seding SwarmCmd: {e}");
+    // wait until we discover atleast one peer
+    peer_dicovered_rx.await?;
+    info!("Discovered a Peer");
+    // todo: sometimes, the node might query the network before it adds a peer to the DHT. The
+    // PeerDiscoverd event is triggered when it adds the peer to the DHT, but the op might fail and
+    // there is no way to confirm it since `RoutingUpdate` is private/no debug impl.
+    // Hence sleep for sometime before querying the network
+    thread::sleep(time::Duration::from_millis(100));
+
+    if let Some(files_path) = opt.upload_chunks {
+        for entry in WalkDir::new(files_path).into_iter().flatten() {
+            if entry.file_type().is_file() {
+                let file = fs::read(entry.path())?;
+                let chunk = Chunk::new(Bytes::from(file));
+                let xor_name = chunk.name();
+                info!(
+                    "Storing file {:?} with xorname: {xor_name:x}",
+                    entry.file_name()
+                );
+                storage.store(&chunk).await?;
+                // store the name as key in the network
+                // Advertise oneself as a provider of the file on the DHT.
+                network_api.store_data(*xor_name).await?;
```

Comment:
And then we store data in the network?

Isn't it so that we do what the comment says, we register ourselves as a provider of the given key?

So... `register_as_provider_of`, was the first that popped into my head. Can surely be more succinctly expressed with a few thought cycles more.

Reply:
This feels like the wrong way round from my understanding...

Im not sure we want to be using the `provider` stuff at all? We just want to send the data to the closest nodes, and they store it. Or?

Reply:
Yes, this might not be what we want to do. Adding some todos here.